### PR TITLE
Update Docs: CountersAndLists to Beta

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Work on #3775 

**Special notes for your reviewer**:

### CODE REVIEW

File: examples/gameserver.yaml [hunk 1]
```diff
@@ -78,33 +78,33 @@ spec:
...
+        count: 1
+    lists:  # lists are lists of values stored against this GameServer that can be added and deleted from. Keys must be declared at GameServer creation time.
+      players:  # an empty list, with a capacity set to 10.
+        capacity: 10  # capacity value, defaults to 1000.
+      rooms:
+        capacity: 333
+        values:  # initial set of values in a list.
+          - room1
+          - room2
+          - room3
```
Comment (lacroixthomas) [inline examples/gameserver.yaml:107]: Shouldn't that be at the `spec` level instead (one level below) ? It seems to be added at the sdkServer level here 
https://agones.dev/site/docs/reference/agones_crd_api_reference/#agones.dev/v1.GameServerSpec

Comment (Kalaiselvi84) [reply examples/gameserver.yaml]: Thanks @lacroixthomas! I've fixed the alignment for counters and lists.

File: examples/gameserverallocation.yaml [hunk 2]
```diff
@@ -85,35 +85,35 @@ spec:
...
+        action: Increment  # Either "Increment" or "Decrement" the Counter’s Count.
+        amount: 1  # Amount is the amount to increment or decrement the Count. Must be a positive integer.
+        capacity: 5  # Amount to update the maximum capacity of the Counter to this number. Min 0, Max int64.
+    # List actions to perform during allocation. Optional.
+    lists:
+      players:
+        addValues:  # appends values to a List’s Values array. Any duplicate values will be ignored
+          - x7un
+          - 8inz
+        capacity: 40  # Updates the maximum capacity of the Counter to this number. Min 0, Max 1000.
```
[the remark this answers is not in the document]
Comment (Kalaiselvi84) [reply examples/gameserverallocation.yaml]: I've fixed the alignment.

File: site/content/en/docs/Reference/gameserver.md [hunk 3]
```diff
@@ -153,6 +154,153 @@ The `spec` field is the actual GameServer specification and it is composed as fo
...
+  lists: # lists are lists of values stored against this GameServer that can be added and deleted from. Keys must be declared at GameServer creation time.
+    players: # an empty list, with a capacity set to 10.
+      capacity: 10 # capacity value, defaults to 1000.
+    rooms:
+      capacity: 333
+      values: # initial set of values in a list.
+        - room1
+        - room2
+        - room3
+  #
```
Comment (igooch) [inline site/content/en/docs/Reference/gameserver.md:246]: Nit:

```suggestion
```

**Review by igooch — APPROVED** [review]
LGTM

**Review by igooch — COMMENTED** [review]
Also update the lines as well:

```
10. Alpha().SetCounterCount()
11. Alpha().IncrementCounter()
12. Alpha().DecrementCounter()
13. Alpha().SetCounterCapacity()
14. Alpha().AppendListValue()
15. Alpha().DeleteListValue()
16. Alpha().SetListCapacity()
```

File: site/content/en/docs/Reference/gameserver.md (hunk 3 above)
Comment (Kalaiselvi84) [reply site/content/en/docs/Reference/gameserver.md]: I've removed this line

**Review by zmerlynn — APPROVED** [review]

**Review by lacroixthomas — APPROVED** [review]

**Review by igooch — APPROVED** [review]
LGTM

**Pull request merged**